feat(datepicker): add suppression of validation for #3836

Blackbaud-PatrickOFriel opened the pull request:
This adds the `$datepickerSuppressError` value provider which has an `isSuppressed` property to allow the option to suppress the `$log` error in `DatepickerController`. This was made to address #3836

wesleycho commented on src/datepicker/datepicker.js:
```diff
@@ -1,5 +1,9 @@
 angular.module('ui.bootstrap.datepicker', ['ui.bootstrap.dateparser', 'ui.bootstrap.position'])
 
+.value('$datepickerSuppressError', { 
```
Just make it a boolean - no need for an object here.

wesleycho commented:
Just make all changes related to my comment and this PR should be good to get in.

Blackbaud-PatrickOFriel commented:
The only reason I made it an object is because when it's a literal, changes you make upon injection don't take effect (setting `$datepickerSuppressError` to `true`). You could only change the value in the config phase using `$provide`. If that's the route I should take, then I'll have to make the test a bit different since you can't use `$provide` after you've used `inject`.

wesleycho commented:
I think that is fine - there really isn't a great use case for toggling suppression on or off once the app is running.

Blackbaud-PatrickOFriel commented:
Okay I'll go ahead and make the change then

Blackbaud-PatrickOFriel commented:
So in order to test properly I had to wrap the other tests in their own describe block so that I could override the `$provide` for my test cases before the `inject` function was ever called. And now the diff tool makes it look like I deleted everything and added it back...

wesleycho commented:
That's perfectly fine

wesleycho commented:
Can you squash your commits into one?

Blackbaud-PatrickOFriel commented:
Squash complete

wesleycho commented:
LGTM - merging shortly. Thanks for the work!

Blackbaud-PatrickOFriel commented:
No problem! Thanks for this entire library, it's great!
